kornia/augmentation/_2d/intensity/motion_blur.py

```diff
@@ -83,7 +84,7 @@ def __init__(
     def apply_transform(
         self, input: torch.Tensor, params: Dict[str, torch.Tensor], transform: Optional[torch.Tensor] = None
     ) -> torch.Tensor:
-        kernel_size: int = cast(int, params["ksize_factor"].unique().item())
+        kernel_size: int = cast(int, random.choice(params["ksize_factor"]).item())
```

Comment:
Theoretically, all randomness shall happen in random_generators, and be controlled under the torch random state.

Also, you may need to update the doc.

Reply:
+1